### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: MultiTable/src/Utils.js [hunk 1]
```diff
@@ -14,10 +14,17 @@
...
     const inv = {
+      // has all the samples in the subject hierarchy keyed off of sample name
       samples: {},
+
+      /*
+       * has all the subjects keyed off of subject absolutepath. The entire
+       * subject hierarchy, including the samples are part of each subject
+       * TODO: Including the subject hierarchy for each of the subject seems
+       * unncecessary. Just include the subject and its samples here.
+       */
```
Comment (shriramshankar) [inline MultiTable/src/Utils.js]: This TODO comment makes sense correct?

File: MultiTable/src/RealtimeChangeHandler.js [hunk 2]
```diff
@@ -74,9 +74,13 @@ function onSubjectAdd(data, subject) {
     if (!subjectGroup) {
       const parentPath = subject.parentAbsolutePath;
       const grandparentGroup = data.getParentGroupForAbsolutePath(parentPath);
-      if (!grandparentGroup) return;
+      if (!grandparentGroup) {
```
Comment (iamigo) [inline MultiTable/src/RealtimeChangeHandler.js]: FYI it's OK to leave these as single line per style guide

File: MultiTable/src/SubjectGroups.js [hunk 3]
```diff
@@ -41,14 +46,23 @@ function jsonToSubjectGroups(json) {
...
+     */
     groups[grpName.toLowerCase()].addSubject(s);
   });
 
   // Add Samples. Ignore sample if its name is invalid.
   d3c.values(inv.samples).forEach((s) => {
     const grpName = deriveGroupName(s.name);
     try {
+      /*
+       * The samples are added to the subjetGroup one level up the hierarchy.
```
Comment (iamigo) [inline MultiTable/src/SubjectGroups.js]: subjectGroup spelling

File: MultiTable/src/SubjectGroups.js [hunk 4]
```diff
@@ -41,14 +46,23 @@ function jsonToSubjectGroups(json) {
       groups[grpName.toLowerCase()] = new SubjectGroup(grpName,
         inv.subjects[grpName.toLowerCase()]);
     }
-
+    /*
+     * The subjects are added to the subjetGroup one level up the hierarchy.
+     * for e.g if the subject is 'US.CA.SFO', the subject will be added to the
```
Comment (iamigo) [inline MultiTable/src/SubjectGroups.js]: `e.g.` (not `for e.g`)

File: MultiTable/src/SubjectGroups.js [hunk 5]
```diff
@@ -41,14 +46,23 @@ function jsonToSubjectGroups(json) {
...
     groups[grpName.toLowerCase()].addSubject(s);
   });
 
   // Add Samples. Ignore sample if its name is invalid.
   d3c.values(inv.samples).forEach((s) => {
     const grpName = deriveGroupName(s.name);
     try {
+      /*
+       * The samples are added to the subjetGroup one level up the hierarchy.
+       * for e.g if the sample is 'US.CA.SFO|delay', the samples is added to the
```
Comment (iamigo) [inline MultiTable/src/SubjectGroups.js]: `e.g.` (not `for e.g`)

Comment (iamigo) [inline MultiTable/src/SubjectGroups.js]: "the samples is added" --> "samples are added"

File: MultiTable/src/Utils.js (hunk 1 above)
Comment (iamigo) [reply MultiTable/src/Utils.js]: If we never use the deep hierarchy from here, then yeah (and please create new user story for this)

File: MultiTable/test/SubjectGroup.js [hunk 6]
```diff
@@ -27,6 +35,77 @@ describe('SubjectGroups Tests', () => {
...
+      expect(Object.keys(samples).length).to.equal(1);
+      expect(Object.keys(aspects).length).to.equal(1);
+
+      expect(subjects[subject.absolutePath.toLowerCase()]).to.eql(subject);
+      expect(samples[sample.name.toLowerCase()]).to.eql(sample);
+      expect(aspects[sample.aspect.name.toLowerCase()]).to.eql(sample.aspect);
+
+    });
+
+    it('calling updateSubject should the updated subject in SubjectGroup', () => {
```
Comment (iamigo) [inline MultiTable/test/SubjectGroup.js]: Missing a word?

File: MultiTable/test/SubjectGroup.js [hunk 7]
```diff
@@ -27,6 +35,77 @@ describe('SubjectGroups Tests', () => {
...
+    it('calling deleteSubject should delete the subject in subjectGroup', () => {
+      const subjectGroup = new SubjectGroup(parentSubject.parentAbsolutePath,
+        parentSubject);
+
+      subjectGroup.addSubject(subject);
+      subjectGroup.removeSubject(subject);
+      expect(subjectGroup.subjects).to.deep.equal({})
+    });
+
+    it('Samples and aspects should be added to their respective  data ' +
```
Comment (iamigo) [inline MultiTable/test/SubjectGroup.js]: two spaces before `data`?

File: MultiTable/src/SubjectGroups.js [hunk 8]
```diff
@@ -14,13 +14,18 @@ const d3c = require('./lib/d3-collection.v1.min.js');
 const SubjectGroup = require('./SubjectGroup');
 const Utils = require('./Utils');
 
+/*
+ * Return the absolutePath one level up from the absolutePath or sample name
+ * specified. Just the first part of the condition is required, other probably not
```
Comment (jgraff2) [inline MultiTable/src/SubjectGroups.js]: what do you mean by this?

Comment (shriramshankar) [reply MultiTable/src/SubjectGroups.js]: `Just the first part of the condition is required, other probably not` 
I just wrote this for myself inititally when I saw line 22 but later realized that I was wrong. I will remove this comment.

**Comment by shriramshankar** [comment]
I will create the zip once the review is done.

**Comment by shriramshankar** [comment]
@iamigo @jgraff2 PR updated with the requested changes.

File: MultiTable/src/RealtimeChangeHandler.js (hunk 2 above)
Comment (jgraff2) [reply MultiTable/src/RealtimeChangeHandler.js]: I think it's more readable as a single line for simple returns like this.

Comment (shriramshankar) [reply MultiTable/src/RealtimeChangeHandler.js]: Most of the if blocks in this repo and other refocus related repos have brackets and I want to make sure it is consistent everywhere.

**Comment by iamigo** [comment]
@shriramshankar add zip to dist

**Comment by shriramshankar** [comment]
@iamigo Already added the zip to the dist. Also, tested this dist in sandbox and updated the 'beta SR Grid'.

**Pull request merged**